From 5de06e39b421b2db74d30eb0ea76f8b754f8546b Mon Sep 17 00:00:00 2001
From: Alberto Garcia <berto@igalia.com>
Date: Tue, 2 Jun 2026 07:24:38 -0700
Subject: [PATCH] REGRESSION(313606@main): Fails to build with system malloc
 https://bugs.webkit.org/show_bug.cgi?id=316083

Reviewed by Fujii Hironori.

Don't add bmalloc_CopyHeaders to LLIntOffsetsExtractor_DEPENDENCIES if
USE_SYSTEM_MALLOC is on.

* Source/JavaScriptCore/CMakeLists.txt:

Canonical link: https://commits.webkit.org/314364@main
---
 Source/JavaScriptCore/CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/Source/JavaScriptCore/CMakeLists.txt b/Source/JavaScriptCore/CMakeLists.txt
index e65f1d4237d9..2de9aa7792e8 100644
--- a/Source/JavaScriptCore/CMakeLists.txt
+++ b/Source/JavaScriptCore/CMakeLists.txt
@@ -427,8 +427,10 @@ set(LLIntSettingsExtractor_DEPENDENCIES
     JavaScriptCore_CopyHeaders
     JavaScriptCore_CopyPrivateHeaders
     WTF_CopyHeaders
-    bmalloc_CopyHeaders
 )
+if (NOT USE_SYSTEM_MALLOC)
+    list(APPEND LLIntSettingsExtractor_DEPENDENCIES bmalloc_CopyHeaders)
+endif ()
 WEBKIT_EXECUTABLE(LLIntSettingsExtractor)
 
 # LLIntSettingsExtractor target needs to have a direct or indirect
@@ -454,8 +456,10 @@ set(LLIntOffsetsExtractor_DEPENDENCIES
     JavaScriptCore_CopyHeaders
     JavaScriptCore_CopyPrivateHeaders
     WTF_CopyHeaders
-    bmalloc_CopyHeaders
 )
+if (NOT USE_SYSTEM_MALLOC)
+    list(APPEND LLIntOffsetsExtractor_DEPENDENCIES bmalloc_CopyHeaders)
+endif ()
 WEBKIT_EXECUTABLE(LLIntOffsetsExtractor)
 
 # The build system will execute asm.rb every time LLIntOffsetsExtractor's mtime is newer than
